velox/functions/lib/string/StringCore.h

```diff
@@ -311,6 +312,41 @@ cappedByteLengthUnicode(const char* input, size_t size, int64_t maxChars) {
   return utf8Position;
 }
 
+constexpr size_t long_string_find(
+    std::string_view str, const std::string_view substr, size_t start_pos
+) noexcept
+{
+    const size_t str_len = str.size();
+    const size_t substr_len = substr.size();
+    if (substr_len == 0)
+        return start_pos <= str_len ? start_pos : std::string_view::npos;
+
+    if (start_pos >= str_len)
+        return std::string_view::npos;
+
+    const char first_char = substr.data()[0];
+    const char* current = str.data() + start_pos;
+    const char* const end = str.data() + str_len;
+    size_t remaining = str_len - start_pos;
+
+    while (remaining >= substr_len)
```

Comment:
please also review our contributing and coding style guidelines:

https://github.com/facebookincubator/velox/blob/main/CONTRIBUTING.md
https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md